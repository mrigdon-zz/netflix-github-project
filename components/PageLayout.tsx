import Head from "next/head";
import React, { useState } from "react";
import Label from "../components/Label";
import LocalePicker from "../components/LocalePicker";
import styles from "../styles/PageLayout.module.css";
import { Locale, LocaleContext } from "../utils/i18n";

export default function PageLayout(props: {
  locale: Locale;
  children: React.ReactNode;
}) {
  const [locale, setLocale] = useState(props.locale);
  return (
    <LocaleContext.Provider value={{ locale, setLocale }}>
      <Head>
        <title>Github Org Repo Explorer</title>
        <link rel="icon" href="/favicon.ico" />
      </Head>

      <header className={styles.header}>
        <div className={styles.headerContent}>
          <div className={styles.nav}>
            <img
              className={styles.logo}
              src="/logo.png"
              alt="Netflix"
              height={40}
            />

            <a href="/" className={styles.navLink}>
              <Label name="home" />
            </a>
          </div>

          <LocalePicker />
        </div>
      </header>

      <main className={styles.container}>{props.children}</main>
    </LocaleContext.Provider>
  );
}
