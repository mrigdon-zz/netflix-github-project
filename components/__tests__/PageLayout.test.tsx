import React from "react";
import { createRenderer } from "react-test-renderer/shallow";
import PageLayout from "../PageLayout";

describe("PageLayout", () => {
  it("renders the page", () => {
    const renderer = createRenderer();
    renderer.render(<PageLayout locale="en">hello</PageLayout>);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
