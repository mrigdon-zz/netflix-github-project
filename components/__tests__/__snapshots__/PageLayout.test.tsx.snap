// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageLayout renders the page 1`] = `
<Context.Provider
  value={
    Object {
      "locale": "en",
      "setLocale": [Function],
    }
  }
>
  <Head>
    <title>
      Github Org Repo Explorer
    </title>
    <link
      href="/favicon.ico"
      rel="icon"
    />
  </Head>
  <header
    className="header"
  >
    <div
      className="headerContent"
    >
      <div
        className="nav"
      >
        <img
          alt="Netflix"
          className="logo"
          height={40}
          src="/logo.png"
        />
        <a
          className="navLink"
          href="/"
        >
          <Label
            name="home"
          />
        </a>
      </div>
      <LocalePicker />
    </div>
  </header>
  <main
    className="container"
  >
    hello
  </main>
</Context.Provider>
`;
